client/app/dedicatedCloud/mailing-list/subscribe/dedicatedCloud-mailing-list-subscribe.controller.js

```diff
@@ -1,21 +1,69 @@
-angular.module("App").controller("DedicatedCloudMailingCtrl", ($scope, dedicatedCloudMailingList, Alerter) => {
-    $scope.load = {
-        loading: false
-    };
-
-    $scope.mailing = {
-        pcc: $scope.currentActionData,
-        email: null
-    };
-
-    $scope.subscribe = function () {
-        $scope.load.loading = true;
-        dedicatedCloudMailingList.postMailingList($scope.mailing.email, $scope.mailing.pcc)
-            .then(() => Alerter.success($scope.tr("dedicatedCloud_subscribe_mailing_step2_success", $scope.mailing.email), "dedicatedCloud"))
-            .catch((err) => Alerter.alertFromSWS($scope.tr("dedicatedCloud_subscribe_mailing_step2_error", $scope.mailing.email), err, "dedicatedCloud"))
-            .finally(() => {
-                $scope.resetAction();
-                $scope.load.loading = false;
-            });
-    };
+angular.module("App").controller("DedicatedCloudMailingCtrl", class DedicatedCloudMailingCtrl {
+
+    constructor ($uibModalInstance, User, dedicatedCloudMailingList, Alerter, translator) {
+        // dependencies injections
+        this.$uibModalInstance = $uibModalInstance;
+        this.User = User;
+        this.dedicatedCloudMailingList = dedicatedCloudMailingList;
+        this.Alerter = Alerter;
+        this.translator = translator;
+
+        // controller attributes
+        this.loading = {
+            init: false,
+            subscribe: false
+        };
+
+        this.canSuscribe = true;
+        this.model = {
+            email: null
+        };
+
+        this.subscribeToMailingList = this.onWidzardFinish.bind(this);
+    }
+
+    /* =============================
+    =            EVENTS            =
+    ============================== */
+
+    onWidzardFinish () {
+        this.loading.subscribe = true;
+
+        return this.dedicatedCloudMailingList.postMailingList(this.model.email, "pcc@ml.ovh.net").then(() =>
+            this.Alerter.success(this.translator.tr("dedicatedCloud_subscribe_mailing_step2_success", this.model.email), "dedicatedCloud")
+        ).catch((error) =>
+            this.Alerter.error([this.translator.tr("dedicatedCloud_subscribe_mailing_step2_error", this.model.email), _.get(error, "message")].join(". "), "dedicatedCloud_alert")
+        ).finally(() => {
+            this.$uibModalInstance.close();
```

Comment:
You should use UI-Router to change the state ... and the modal will be closed :)

Modal or not should be managed by only one parameter -> `layout: 'modal'`